src/test/java/core/basesyntax/service/implementation/ReportGenerationServiceImplTest.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.service.implementation;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerationService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ReportGenerationServiceImplTest {
+    private static String SEPARATOR = System.lineSeparator();
+    private static ReportGenerationService reportGenerationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        reportGenerationService = new ReportGenerationServiceImpl();
+    }
+
+    @Test
+    void generate_validData_ok() {
+        Storage.STORAGE.put("apricot", 15);
+        String report = reportGenerationService.generate();
+        String actualReport = "fruit,quantity" + SEPARATOR
+                + "apricot,15" + SEPARATOR;
+        assertEquals(actualReport, report);
+    }
+
+    @Test
+    void generate_negativeData_ok() {
```

Comment:
If expect that exception is thrown in your test case than this is notOk case.
```suggestion
    void generate_negativeData_notOk() {
```